ios/Exponent/Kernel/Views/EXAppViewController.m

```diff
@@ -362,6 +393,11 @@ - (void)appLoader:(EXAppLoader *)appLoader didLoadOptimisticManifest:(NSDictiona
     [[EXKernel sharedInstance].browserController addHistoryItemWithUrl:appLoader.manifestUrl manifest:manifest];
   }
   [self _rebuildBridge];
+
+  NSString *loadingViewText = [self _loadingViewTextForStatus:appLoader.remoteUpdateStatus];
+  if (loadingViewText) {
+    [self.appLoadingProgressWindowController updateStatusWithText:loadingViewText];
```

Comment:
Maybe, in order to prevent this `if`, you can make `updateStatusWithText` accept nullable argument?

Reply:
Seems better not to call it if the operation doesn't need to happen. Make the code flatter instead of nested. Setting to nil is also ambiguous -- does it clear the text or does only the empty string do that?